server/start.go

```diff
@@ -313,30 +310,47 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 	}
 
 	genDocProvider := node.DefaultGenesisDocProviderFunc(cfg)
-	tmNode, err := node.NewNode(
-		cfg,
-		pvm.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile()),
-		nodeKey,
-		proxy.NewLocalClientCreator(app),
-		genDocProvider,
-		node.DefaultDBProvider,
-		node.DefaultMetricsProvider(cfg.Instrumentation),
-		ctx.Logger.With("server", "node"),
+	var (
+		tmNode   *node.Node
+		gRPCOnly = ctx.Viper.GetBool(srvflags.GRPCOnly)
 	)
-	if err != nil {
-		logger.Error("failed init node", "error", err.Error())
-		return err
-	}
+	if gRPCOnly {
+		ctx.Logger.Info("starting node in query only mode; Tendermint is disabled")
+		config.GRPC.Enable = true
+		config.JSONRPC.EnableIndexer = false
+	} else {
+		tmNode, err = node.NewNode(
+			cfg,
+			pvm.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile()),
+			nodeKey,
+			proxy.NewLocalClientCreator(app),
+			genDocProvider,
+			node.DefaultDBProvider,
+			node.DefaultMetricsProvider(cfg.Instrumentation),
+			ctx.Logger.With("server", "node"),
+		)
+		if err != nil {
+			logger.Error("failed init node", "error", err.Error())
+			return err
+		}
 
-	if err := tmNode.Start(); err != nil {
-		logger.Error("failed start tendermint server", "error", err.Error())
-		return err
+		if err := tmNode.Start(); err != nil {
```

Comment:
> tac0turtle: Are the queries from Tendermint supported in this mode? Stuff like block?

> alexanderbez: No, this is gRPC only mode. The doc and implementation states Tendermint is disabled (i.e. not started). Tendermint cannot be supported.

Is it ok to disable tendermint? 
For example: [method GetCoinbaseAddress](https://github.com/evmos/ethermint/blob/db94d8ec231585b32adf026b7f39434c810c4079/x/evm/keeper/state_transition.go#L524), which was called by [method EvmConfig](https://github.com/evmos/ethermint/blob/db94d8ec231585b32adf026b7f39434c810c4079/x/evm/keeper/state_transition.go#L47), needs validator addr of the context block, how can you get that without tendermint?

Reply:
For some queries, you'd need to run the gRPC with Tendermint